app/controllers/tickets_controller.rb

```diff
@@ -37,8 +39,8 @@ def destroy
   end
 
   def state # change ticket state
-    unless current_user.is_admin || current_user.worker.role == "Manager"
-      if current_user.worker.id == @ticket.worker_id
+
+      if current_user.is_admin || current_user.worker.role == "Manager" || current_user.worker.id == @ticket.worker_id
```

Comment:
this logic does not belong to controller layer.
There are few ways to refactor it, implement one of:
1) Create on user model method like `can_update_ticket_state?(ticket)` which will contain this logic
2) Create on ticket mode method `state_can_be_change_by?(user)`
3) Use https://github.com/varvet/pundit gem which is very popular abstraction layer for permissions 
My advice is to implement point 3